libraries/base/Data/Semigroup.hs

```diff
@@ -18,21 +18,36 @@
 -- Maintainer  :  libraries@haskell.org
 -- Stability   :  provisional
 -- Portability :  portable
+-- 
+-- A type @a@ is a 'Semigroup' if it provides an associative function ('<>')
+-- that lets you combine any two values of type @a@ into one. 
 --
--- In mathematics, a semigroup is an algebraic structure consisting of a
--- set together with an associative binary operation. A semigroup
--- generalizes a monoid in that there might not exist an identity
--- element. It also (originally) generalized a group (a monoid with all
--- inverses) to a type where every element did not have to have an inverse,
--- thus the name semigroup.
```

Comment:
Can we please not remove useful documentation?

Reply:
The fact that it generalizes the concept of a Monoid is mentioned below under "Theoretical Context". As for the rest: I am having a very hard time understanding how this is relevant or helpful at all.

How is the fact that it originally generalized a group, which is just a  monoid with all inverses (i actually don't understand what that even means) useful documentation? Who would miss this piece of information if it was not there and/or be less able to make use of Semigroup? Isn't this the kind of information that the avid user can look for at nlab or wikipedia?

if you and others really insist on including the information on how the name came to be then I suppose this can be added under "Theoretical Context", but please don't make me start documentation with "In mathematics" and explanations on the origins of the name. Shouldn't documentation focus first on the *purpose* and how it can be used and applied?